06-building-a-program-start-to-finish/solving-the-problem/output/InvertedIndex.js

```diff
@@ -0,0 +1,242 @@
+      }
+    });
+
+    return this.file;
+  }
+
+  /**
+   * @desc tokenize: Returns an array of valid words in all text fields of file.
+     Also returns an array of valid words in a single string.
+   * @param {(Object | String)} file - object or string to be tokenized
+   - the object whose texts are to be tokenized or string to tokenize
+   * @return {array} - array of valid words
+   */
+  tokenize(file) {
+    const validWord = /[a-zA-Z]+/g;
+    this.file = file;
+    let words = '';
+    let found;
+    const validWordsArray = [];
+
+    if (typeof (this.file) === 'object') {
+      for (let i = 0; i < this.file.length; i += 1) {
+        words = words.concat(`${this.file[i].text} `);
+      }
+
+      while (found = validWord.exec(words)) {
+        validWordsArray.push(found[0]);
+      }
+
+      // make the words in array unique and sort them
+      return [...new Set(validWordsArray)].sort();
+    }
+
+    while (found = validWord.exec(this.file)) {
+      validWordsArray.push(found[0]);
+    }
+
+    return validWordsArray;
+  }
+
+  /**
+   * @desc getTitles: Gets the titles of documents in uploaded file
+   * @param {Object} file - the object whose document titles you want to get
+   * @return {array} - returns an array of document titles
+   */
+  getTitles(file) {
+    this.file = file;
+    const titles = [];
+    const texts = [];
+
+    for (let i = 0; i < this.file.length; i += 1) {
+      let fileTitle = this.file[i].title;
+
+      if (texts.indexOf(this.file[i].text) === -1) {
+        if (titles.indexOf(fileTitle) !== -1) {
+          fileTitle += ' - Copy';
+        }
+        titles.push(fileTitle);
+      } else if (titles.indexOf(fileTitle) === -1) {
+        titles.push(fileTitle);
+      }
+
+      texts.push(this.file[i].text);
+    }
+
+    return titles;
+  }
+
+  /**
+   * @desc lowerDocText: Turns the text fields of documents to lower case
+   * @param {Object} file - the object whose document texts you want to lower
+   * @return {Object} - returns the object with its text fields lowered
+   */
+  lowerDocText(file) {
+    this.file = file;
+    this.file.forEach((doc) => {
+      const eachDoc = doc;
+      eachDoc.text = eachDoc.text.toLowerCase();
+    });
+
+    return this.file;
+  }
+
+  /**
+   * @desc createIndex:
+   Creates indices (words and indices of documents they appear in) for a file
+   * @param {string} fileName - the name of the file you want to index
+   * @param {Object} fileContent - object representing the contents of file
+   * @throws - throws an error if object does not conform to required structure
+   * @return {array} - returns an array
+    containing indices for file and titles of documents in file
+   */
+  createIndex(fileName, fileContent) {
+    let fileRead;
+    try {
+      fileRead = this.lowerDocText(fileContent);
+    } catch (err) {
+      throw new Error('This file\'s structure is ' +
+      'invalid. Only array of objects are allowed.');
```

Comment:
Eslint raised an error that the line was too long. This is the solution recommended on the website. Also, template literals didn't behave as I expected when I tried using them. My tests failed because.